src/pages/academy/game/Game.tsx

```diff
@@ -50,9 +52,93 @@ function Game() {
     }
   }, [session, achievements, goals]);
 
+  // This is a custom hook imported from @mantine/hooks that handles the fullscreen logic
+  // It returns a ref to attach to the element that should be fullscreened,
+  // a function to toggle fullscreen and a boolean indicating whether the element is fullscreen
+  const {
+    ref: fullscreenRef,
+    toggle: toggleFullscreen,
+    fullscreen: isFullscreen
+  } = useFullscreen<HTMLDivElement>();
+
+  // This function is a wrapper around toggleFullscreen that also locks the screen orientation
+  // to landscape when entering fullscreen and unlocks it when exiting fullscreen
+  const enhancedToggleFullscreen = async () => {
+    toggleFullscreen();
+
+    if (window.screen.orientation) {
+      if (!isFullscreen) {
+        window.screen.orientation.lock('landscape');
+      } else {
+        window.screen.orientation.unlock();
+      }
+    }
+  };
+
+  const gameDisplayRef = React.useRef<HTMLDivElement | null>(null);
+
+  // This function sets the gameDisplayRef and also calls the ref callback from useFullscreen
+  // to attach the fullscreen logic to the game display element
+  const setGameDisplayRefs = React.useCallback(
+    (node: HTMLDivElement | null) => {
+      // Refs returned by useRef()
+      gameDisplayRef.current = node;
+
+      // Ref callback from useFullscreen
+      fullscreenRef(node);
+    },
+    [fullscreenRef]
+  );
+
+  // Logic for the fullscreen button to dynamically adjust its size, position and padding
+  // based on the size of the game display.
+  const [iconSize, setIconSize] = React.useState(0);
+  const [iconLeft, setIconLeft] = React.useState('0px');
+  const [iconPadding, setIconPadding] = React.useState('0px');
+
+  React.useEffect(() => {
+    const handleResize = () => {
```

Comment:
Please don't put this function definition inside the `useEffect` body, coupled with the resize observer below it results in so many unnecessary function objects being recreated, degrading performance.